### DISCUSSION TITLE
Using "Show text" node (pythongosssss) as HUD display for "Switch (Any)" Node (Impact Pack)

### DISCUSSION THREAD

**Discussion by JorgeR81** [comfyanonymous/ComfyUI#2294] [Q&A] [answered]
The "Switch (Any)" Node (Impact Pack) has an "selected_label" output, so I'm trying to use the "Show text" node (pythongosssss) as an HUD display

This works very well at first, but when I open a saved workflow, the text in the "Show text" node is blank. And it only shows up again when I change a switch input and run a prompt. 

Has anyone tried this ?
Is this the way these nodes are supposed to work, or is something not working well for me?
It would be more useful if the text was always visible, especially for complex workflows, with many switches. 

I'm not sure if the issue is related to the Comfy UI or the custom nodes, so I posted here.  

Thanks 

![swtext](https://github.com/comfyanonymous/ComfyUI/assets/72734038/59fc6b7d-381b-4585-a156-5af4bda6368e)

**Comment by ltdrdata** [comment, marked as answer]
In the workflow, information is stored before the prompt is executed. The `Show text` node, on the other hand, updates its information after the workflow has been executed.

To implement this functionality, it is necessary to create a specialized custom node for the `Switch (Any)` that operates independently during workflow execution.

**Comment by JorgeR81** [comment]
OK. thanks.
I'm marking my question as resolved.
But I'm hoping you're able to add this feature.

The switch node allows me to create more complex workflows, that make it easier to try new things. 
I'm having very interesting results when using 2 sets of prompts. With this I can try many variations, just on the flip of a switch!
I also like to put switches in all nodes that go into the "EditBasicPipe'' node, so I end up with a lot of Switches.

A selected label display on the switch node itself would make it easier to check my current workflow configuration.

I'm leaving the discussion open, in case someone else wants to comment. 
But, feel free to close it, if you think is best, since this is more related to the custom nodes.

![psw2](https://github.com/comfyanonymous/ComfyUI/assets/72734038/0c69c621-29e5-432e-b745-3f987fab7ac5)

**Comment by JorgeR81** [comment]
Perhaps it would be more straightforward to just display the input label text in the selection widget ( see edited image ), instead of creating an extra area on the node to display the text. 
( although I don't know if this would be harder or easier to implement )
If you want I can open a feature request on the Impact Pack page, to keep things more organized.
But I recognize that this should be a low priority, since the current node works fine. 

![sw2](https://github.com/comfyanonymous/ComfyUI/assets/72734038/ee56953f-0037-4ba4-a3e4-7d498933c399)
